arches/app/models/resource.py

```diff
@@ -189,22 +189,31 @@ def bulk_save(resources):
             resource.tiles = resource.get_flattened_tiles()
             tiles.extend(resource.tiles)
 
-        # need to save the models first before getting the documents for index
+        # need to handle if the bulk load is appending tiles to existing resources/
+        existing_resources = Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources])
```

Comment:
On line 193 I would suggest the following changes for performance:
``existing_resources = set(Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]).values_list('resourceinstanceid', flat=True))``
This would obviate the need for the work on line 193

Reply:
I've changed this and am testing

Reply:
this failed testing so reverted - see comment below